CONTRIBUTING.md

```diff
@@ -1,191 +1,5 @@
-
-- Modify the documentation as necessary.
-
-- Verify the entire CI process (building and testing) works.
-
-While there may be exceptions, the general rule is that all PRs should
-be 100% complete - meaning they should include all test cases and documentation
-changes related to the change.
-
-When ready, if you have not already done so, sign a
-[contributor license agreements](#contributor-license-agreements) and submit
-the PR.
-
-See [Writing Good Pull Requests](https://github.com/istio/istio/wiki/Writing-Good-Pull-Requests) for guidance on creating
-pull requests, and [Reviewing Pull Requests](https://github.com/istio/istio/wiki/Reviewing-Pull-Requests) for the PR review
-we use.
-
-## Issues
-
-[GitHub issues](https://github.com/istio/istio/issues/new/choose) can be used to report bugs or submit feature requests.
-
-When reporting a bug please include the following key pieces of information:
-
-- The version of the project you were using (e.g. version number,
-  or git commit)
-
-- Operating system you are using.
-
-- The exact, minimal, steps needed to reproduce the issue.
-  Submitting a 5 line script will get a much faster response from the team
-  than one that's hundreds of lines long.
-
-## Promote your company on istio.io
-
-If your company supports Istio, you can list it on our [Ecosystem page](https://istio.io/latest/about/ecosystem/).
-We have categories for *providers* (who offer hosted or managed Istio services
-to their customers), *professional services* (who offer implementation support
-or consulting for Istio) and *integrations* (commercial or open source products
-that work with Istio).
-
-To add an entry to this page, edit [data/companies.yml](https://github.com/istio/istio.io/blob/master/data/companies.yml)
-and add your information in the `providers`, `pro_services` or `integrations`
-section, as appropriate. Please add your company logo, preferably in SVG
-format, to [static/logos](https://github.com/istio/istio.io/blob/master/static/logos).
-
-The Istio steering committee reserves the right to remove logos of companies that
-are not demeed to be in good standing in the community.
-
-## Tell the world you're using Istio
-
-Are you running Istio in production?  List yourself on our [Case Studies](https://istio.io/latest/about/case-studies/) page.
-
-If you want to add your logo to our wall of users, add an entry to the `users`
-section in [data/companies.yml](https://github.com/istio/istio.io/blob/master/data/companies.yml), and add your logo,
-preferably in SVG format, to [static/logos](https://github.com/istio/istio.io/blob/master/static/logos).
-
-If you'd like to be interviewed for a full case study, please [create an issue](https://github.com/istio/istio.io/issues/new),
-and we'll be in touch!
+So you want to hack on Istio? Yay! Please refer to Istio's overall
+[contribution guidelines](https://github.com/istio/community/blob/master/CONTRIBUTING.md)
+to find out how you can help.
```

Comment:
This change makes this file self-referential?

I would like the content to remain in a file named CONTRIBUTING.md if possible; that's the standard that most projects use.

Reply:
Looking at the makefile, there is no way to determine which repo is doing the `make update-common`. Thus, I can't stop the copying of the file.

I could remove the file from common-files altogether which would prevent overriding the CONTRIBUTION.md file in the repos and everything stays the same. The negative here is if we ever wanted to update the CONTRIBUTION.md file in the other repos we would have to do it manually (which isn't that hard, just ~10 PRs, but I don't expect to happen all that often, if at all.)

Reply:
Here's my current change (which will get integrated into the common-files) changing to only copy from common-files if the current file has a `refer to Istio` in it (this is part of the current common CONTRIBUTING.md). This allows the CONTRIBUTING.md in istio/community to remain the long version, while the other repos have a shorter version pointing to the istio/community file.

In the future, if we change the common-files CONTRIBUTING.md, we may need to tweak things depending on the text there.